test/onnx/test_fx_to_onnx.py

```diff
@@ -164,6 +168,106 @@ def forward(self, input):
             expected_error_node="aten.convolution.default",
         )
 
+    def test_fake_tensor_mode_simple(self):
+        class Model(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.linear = torch.nn.Linear(2, 2)
+
+            def forward(self, x):
+                out = self.linear(x)
+                return out
+
+        with torch.onnx.enable_fake_mode() as fake_context:
+            x = torch.rand(5, 2, 2)
+            model = Model()
+
+        # Export the model with fake inputs and parameters
+        export_options = ExportOptions(fake_context=fake_context)
+        export_output = torch.onnx.dynamo_export(
+            model, x, export_options=export_options
+        )
+        assert export_output is not None
+        assert export_output.model_proto is not None
+        onnx.checker.check_model(export_output.model_proto, full_check=True)
+        assert len(export_output.model_proto.graph.initializer) == 0
+
+        # Variant 1: Save ONNX proto using Model's state_dict()
+        with tempfile.NamedTemporaryFile(suffix=".onnx") as tmp_onnx_file:
+            model_state_dict = Model().state_dict()  # optional
+            export_output.save(tmp_onnx_file.name, model_state_dict=model_state_dict)
```

Comment:
I actually tested this after https://github.com/pytorch/pytorch/pull/104493, which fixed the names - and it worked. I believe the PR description call this out

Reply:
Cool, so it is by name? #104493 alone doesn't fully restore the name yet.

I have created a follow-up in the stack you might want to check out.

Reply:
Yes, what happens in reality is that `make_fx` lifts initializers as inputs with the correct name (replacing `.` by `_`). Next we create save the actual initializers with the same name. by ONNX spec, when input and initializers are the same, if we don't provide it as input, the internal initializer is used instead. Otherwise, the input data is preferred.

Makes sense?

Reply:
I thought I was clear but now I am confused.

> make_fx lifts initializers as inputs with the correct name (replacing . by _)

If make_fx does this, why does this PR need https://github.com/pytorch/pytorch/pull/104493 ? Can call on Monday if you'd prefer.

Reply:
What I meant is that after #104493, both initializers and graph inputs have matching names, so when initializers are not specified as inputs, the already existing ones in the graph are used instead.

Without #104493, the graph inputs will be `linear_weight` and `linear_bias` while the initializers will be `_param_constant0` and `_param_constant1`.